src/CalcManager/CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+add_library(CEngine
+	CEngine/calc.cpp
+	CEngine/CalcInput.cpp
+	CEngine/CalcUtils.cpp
+	CEngine/History.cpp
+	CEngine/Number.cpp
+	CEngine/Rational.cpp
+	CEngine/RationalMath.cpp
+	CEngine/scicomm.cpp
+	CEngine/scidisp.cpp
+	CEngine/scifunc.cpp
+	CEngine/scioper.cpp
+	CEngine/sciset.cpp
+)
```

Comment:
I'd like to avoid fragmentation between what it means to compile this project with MSVC or Clang.  With MSVC, we fully compile CalcManager and produce CalcManager.lib.  With Clang/CMake, we'd only be compiling CEngine and producing CEngine.lib.  If that's the case, I don't think think putting this file in the CalcManager folder is the right place for it, because it's not a goal for this file to compile CalcManager.  It seems like this should be moved to the CEngine folder.

Separately, I'm curious if your intention is to add support for building other parts of the CalcManager folder in a separate change, or are you only interested in compiling CEngine?  @janisozaur, I'm wondering what your interest is as well.  If people only want CEngine, then we should move it to its own project that we can compile consistently with Clang or MSVC.

Reply:
This should be fixed by the commits 752c3b5 and 8baac74, which package `CalcManager` into a single library and move the inclusions of `CEngine` and `Ratpack` source files into their own directories.

I will, however, leave this conversation open for discussion, since the question remains whether the engine would deserve its own project.

Reply:
@fwcd, @janisozaur, what are your thoughts on us distributing the CalcEngine separately?  CalcManager is very specific to the Windows Calculator app, as it manages the three engine instances needed for Standard, Scientific, and Programmer modes.  We could move CalcManager/UnitConverter out and leave CalcEngine/Ratpak as its own project.  Developers who only want to use CalcEngine would have one project to build in the repo.  We could also make this eaiser by publishing offical distributions of the CalcEngine lib at the same time we publish a release of Windows Calculator to the store.

Reply:
What do you mean by "distributing separately"? Does that imply moving it to a separate repository? Would you create another Visual Studio project for that library? Would there be any benefit on your side?

Reply:
By "distributing separately" I mean we would find a way to distribute the static lib, for example, through a native NuGet package.  Maybe simply tagging the commits is enough and consumers can use vcpkg to incorporate CalcEngine into their projects.  We don't have a concrete plan for this right now, we're just wondering if the interest is there in the community.  In general, it seems that some contributors are only interested in the CalcEngine part of the repo and we benefit if we make it easier for those contributors to build, consume, and improve the CalcEngine.

Reply:
That sounds interesting, but I'm not convinced there's a demand for CalcEngine to be packaged as such a library. Maintaining such a library for external consumers would mean maintaining API and ABI compatibility, fixing the test suite (#340), improving documentation and offering something that GMP, MPFR, Boost.Multiprecision and others don't already offer. I'm not really sold on the idea even though the way to achieving it would address at least some of the issues I raised, but then again I'm not your target audience. Perhaps starting small like tagging commits that you mentioned would suffice for now and when this PR gets merged and there's either https://github.com/Microsoft/calculator/issues/395 implemented or it's equivalent for MSVC, separating out a CalcEngine library should be trivial for those who need it.